calico/common.py

```diff
@@ -67,6 +67,19 @@ def validate_ip_addr(addr, version):
         return False
 
 
+def validate_cidr(cidr, version):
+    """
+    Validates that a CIDR is valid. Returns true if valid, false if
+    not. Version can be "4", "6", None for "IPv4", "IPv6", or "either"
+    respectively.
+    """
+    try:
+        ip = netaddr.IPNetwork(cidr, version=version)
```

Comment:
When we come back to this code it would probably be nicer to just pass the `IPNetwork`s (and `IPAddress`es) around and convert them to strings immediately before programming.